### PR TITLE
Turn dylan-opt into a minor mode

### PR DESCRIPTION
**Pull request by lassik**

### CODE REVIEW

**Comment by lassik** [comment]
Ref issue #46

**Review by housel — APPROVED** [review]

File: dylan-opt.el [hunk 1]
```diff
@@ -72,28 +77,105 @@
...
-(defun dylan-opt--add-overlays (which specs)
-  "Add some Dylan optimization faces to the current buffer.
+(defun dylan-opt--read-all (stream)
+  "Helper to read all Emacs Lisp forms from STREAM."
+  (let ((forms '()))
+    (condition-case _ (while t (push (read stream) forms))
+      (end-of-file (reverse forms)))))
+
+(defun dylan-opt--parse-regions (opt-file)
+  "Parse Dylan optimization regions from OPT-FILE."
```
Comment (cgay) [inline dylan-opt.el:132]: Could use a description of the return value.

File: dylan-opt.el [hunk 2]
```diff
@@ -115,35 +197,31 @@ to add them to."
...
+;;;###autoload
+(define-minor-mode dylan-opt-mode
+  "Toggle Dylan-Opt minor mode for optimization info.
+
+This mode can be used on top of `dylan-mode'.  It shows how
+different regions of a Dylan source file have been optimized by
+the compiler.  The Dylan compiler can produce an optimization dump
+file (*.el) as a byproduct of normal compilation.  Use the
+`dylan-opt' command to feed that file to Emacs and to enable
+highlighting.
```
Comment (cgay) [inline dylan-opt.el:213]: You might mention here that the file is in `_build/build/<library>.el`.

**Review by cgay — APPROVED** [review]

File: dylan-opt.el (hunk 2 above)
Comment (cgay) [reply dylan-opt.el:213]: er, `_build/build/<library>/<filename>.el`

File: dylan-opt.el (hunk 1 above)
Comment (lassik) [reply dylan-opt.el:132]: OK

File: dylan-opt.el (hunk 2 above)
Comment (lassik) [reply dylan-opt.el]: Added.

**Comment by lassik** [comment]
Merge?

**Pull request merged**